skyportal/handlers/api/observation.py

```diff
@@ -443,6 +455,78 @@ def get_observations(
                 intarea = 0.0
 
     total_matches = session.scalar(sa.select(sa.func.count()).select_from(obs_query))
+
+    order_by = None
+    if sort_by is not None:
+        if sort_by == "obstime":
+            order_by = (
+                [Observation.obstime]
+                if sort_order == "asc"
+                else [Observation.obstime.desc()]
+            )
+        elif sort_by == "observation_id":
+            order_by = (
+                [Observation.observation_id]
+                if sort_order == "asc"
+                else [Observation.observation_id.desc()]
+            )
+        elif sort_by == "exposure_time":
+            order_by = (
+                [Observation.exposure_time]
+                if sort_order == "asc"
+                else [Observation.exposure_time.desc()]
+            )
+        elif sort_by == "seeing":
+            order_by = (
+                [Observation.seeing]
+                if sort_order == "asc"
+                else [Observation.seeing.desc()]
+            )
+        elif sort_by == "airmass":
+            order_by = (
+                [Observation.airmass]
+                if sort_order == "asc"
+                else [Observation.airmass.desc()]
+            )
+        elif sort_by == "limmag":
+            order_by = (
+                [Observation.limmag]
+                if sort_order == "asc"
+                else [Observation.limmag.desc()]
+            )
+        elif sort_by == "filt":
+            order_by = (
+                [Observation.filt] if sort_order == "asc" else [Observation.filt.desc()]
+            )
+        elif sort_by == "instrument_name":
+            order_by = (
+                [Observation.instrument_id]
+                if sort_order == "asc"
+                else [Observation.instrument_id.desc()]
+            )
+        elif sort_by == "queue_name":
+            order_by = (
+                [Observation.queue_name]
+                if sort_order == "asc"
+                else [Observation.queue_name.desc()]
+            )
+        elif sort_by == "validity_window_start":
+            order_by = (
+                [Observation.validity_window_start]
+                if sort_order == "asc"
+                else [Observation.validity_window_start.desc()]
+            )
+        elif sort_by == "validity_window_end":
+            order_by = (
+                [Observation.validity_window_end]
+                if sort_order == "asc"
+                else [Observation.validity_window_end.desc()]
+            )
+
```

Comment:
I would put an `else` with raise value error if it doesn't match any of these strings. 
If not, you can skip the `is not None` condition at the beginning as `None` is not equal to any of these. 
Best is to put the `if order by is None` at the first `if` and the `elif` all the strings and leave the last `else` for a wrong string.

Reply:
Good idea.